eta/core/c3d.py

```diff
@@ -0,0 +1,340 @@
+        pool5 = tf.transpose(pool5, perm=[0,1,4,2,3])
+        dense1 = tf.reshape(pool5, [1, _weights['wd1'].get_shape().as_list()[0]]) # Reshape conv3 output to fit dense layer input
+        dense1 = tf.matmul(dense1, _weights['wd1']) + _biases['bd1']
+
+        self.fc6 = tf.nn.relu(dense1, name='fc1') # Relu activation
+        dense1 = tf.nn.dropout(self.fc6, _dropout)
+
+        dense2 = tf.nn.relu(tf.matmul(dense1, _weights['wd2']) + _biases['bd2'], name='fc2') # Relu activation
+        dense2 = tf.nn.dropout(dense2, _dropout)
+
+        # Output: class prediction
+        self.out = tf.matmul(dense2, _weights['out']) + _biases['out']
+
+    def _load_model(self):
+        saver = tf.train.Saver()
+        init = tf.global_variables_initializer()
+        self.sess.run(init)
+        saver.restore(self.sess, self.config.model_path)
+
+
+class C3DFeaturizerConfig(C3DConfig):
+    ''' C3D Featurization configuration settings that works on images'''
+    pass
+
+
+class C3DFeaturizer(Featurizer):
+    def __init__(self, config=None):
+        super(C3DFeaturizer,self).__init__()
+        self.config = config or C3DFeaturizerConfig.default()
+        self.validate(self.config)
+        self.sample_method = self.config.sample_method
+        self.c3d  = None
+
+    def get_first_k_frames(self, inpath, k, out_size):
+        '''Sample first k frames in a video.
+
+        Args:
+            inpath: path to the input video
+            k: number of frames to extract for the input video
+            out_size: size of the output frames
+        '''
+        data = []
+        num_frames = etav.get_frame_count(inpath)
+        assert k < num_frames
+        with etav.VideoProcessor(inpath, "1-%d" % k, out_size=out_size) as p:
+            for img in p:
+                img = etai.resize(img, out_size, out_size)
+                data.append(img)
+        np_arr_data = np.array(data).astype(np.float32)
+        return np_arr_data
+
+    def uniformly_sample_k_frames(self, inpath, k, out_size):
+        '''Uniformly sample k frames, always including the first frame.
+
+        Args:
+            inpath: path to the input video
+            k: number of frames to extract for the input video
+            out_size: size of the output frames
+        '''
+        data = []
+        num_frames = etav.get_frame_count(inpath)
+        assert k < num_frames
+
+        stride = int(math.ceil(1.0 * num_frames / k))
+        rng = range(num_frames)
+        rng = list(rng[::stride])
+        if len(rng) < k:
+            rng.append(num_frames - 1)
+        elif len(rng) > k:
+            rng.pop()
+        assert len(rng) == k
+        with etav.VideoProcessor(inpath, out_size=out_size) as p:
+            for counter, img in enumerate(p):
+                if counter in rng:
+                    img = etai.resize(img, out_size, out_size)
+                    data.append(img)
+        np_arr_data = np.array(data).astype(np.float32)
```

Comment:
This returns a tensor whose first dimension is number of frames. You should follow the typical convention of `m x n x k`, where `k` = number of frames

Reply:
As in the C3D tensorflow model the input is of size [batchsize, number_frames, img_size,  img_size, num_channels]. Should  I convert to the the dimension setting you mentioned and then reshape afterwards?

Reply:
I suppose you can leave it is `k x m x n` for now, but please document this in the docstrings.